### PR TITLE
[IMP] pos_self_order: add time availability on pos category

### PR DESCRIPTION
**Pull request by adgu-odoo**
This commit adds a time availability on pos category for the self order. The time availability able the pos_category to be available only on certain period of time which can be useful for breakfast for example. This period of time must be set by the user in the pos category form.

Task-id: 3617004

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js [hunk 1]
```diff
@@ -85,6 +85,14 @@ export class ProductListPage extends Component {
...
             () => [this.selfOrder.currentCategory]
         );
+
+        onWillStart(() => {
+            this.onWillStart();
+        });
+    }
+
+    onWillStart() {
+        this.selfOrder.updateCategoryList();
```
Comment (caburj) [inline addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js:95]: May I know why is this necessary? The setup of the store already does this calculation.

File: addons/pos_self_order/static/src/app/self_order_service.js [hunk 2]
```diff
@@ -257,6 +250,30 @@ export class SelfOrder extends Reactive {
...
+            .filter((c) => this.productsGroupedByCategory[c.id])
+            .sort((a, b) => categorySorter(a, b, this.config.iface_start_categ_id));
+        this.categoryList = new Set(availableCategories);
+
+        this.availableCategoryListIds = availableCategories
+            .filter((c) => {
+                if (c.hour_after > now || c.hour_until < now) {
+                    return false;
+                }
+                return true;
```
Comment (caburj) [inline addons/pos_self_order/static/src/app/self_order_service.js:267]:
```suggestion
                return now > c.hour_after && now < c.hour_until
```

Or `return !(c.hour_after > now || c.hour_until < now)`

File: addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js (hunk 1 above)
Comment (adgu-odoo) [reply addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js:95]: The setup of the self_order_service is going through the updateCaregoryList one time at the initialization of the self service. Here I want to recompute everytime the product_screen is displayed if the categories are still available or not depending on the time of the day

Comment (adgu-odoo) [reply addons/pos_self_order/static/src/app/pages/product_list_page/product_list_page.js:95]: The setup of the self_order_service is going through the updateCaregoryList one time at the initialization of the self service. Here I want to recompute everytime the product_screen is displayed if the categories are still available or not depending on the time of the day

File: addons/pos_self_order/static/src/app/self_order_service.js (hunk 2 above)
Comment (adgu-odoo) [reply addons/pos_self_order/static/src/app/self_order_service.js:267]: Yep that's better

**Review by caburj — APPROVED** [review]
robodoo r+
